unlock-app/src/hooks/useKeyManager.ts

```diff
@@ -13,6 +17,44 @@ interface Options {
   enabled?: boolean
 }
 
+interface KeyManagerProps {
+  lockAddress: string
+  network: number
+  tokenId: string
+}
+
+export const useKeyManager = ({
+  lockAddress,
+  network,
+  tokenId,
+}: KeyManagerProps) => {
+  const { isLoading, data: manager = null } = useQuery(
+    ['getManagers', lockAddress, network, tokenId],
+    async () => {
+      const subgraph = new SubgraphService()
+      const result = await subgraph.key(
+        {
+          where: {
+            lock: lockAddress,
+            tokenId,
+          },
+        },
+        {
+          network,
+        }
+      )
+
+      // defaults to the onwer when the manager is not set
+      return result?.manager ?? result?.owner ?? ''
+    }
+  )
+
+  return {
+    isLoading,
+    manager,
+  }
```

Comment:
Why not simply use `useKeys()` then return only managers. This way, we maintain a singlular cache for all keys instead of fetching twice for just for getting managers. Less duplication too.

Reply:
refactored with `usekeys` +1